app/src/main/java/org/fcitx/fcitx5/android/input/popup/PopupAction.kt

```diff
@@ -12,17 +12,34 @@ sealed class PopupAction {
 
     abstract val viewId: Int
 
+    enum class PreviewStyle {
+        // Legacy tall preview bubble centered above key
+        Default,
+        // Same size as key and fully placed above it
+        FitAbove,
```

Comment:
If there are hardcoded scale values here: https://github.com/BetterAndBetterII/fcitx5-android/blob/a88fa54ed8f6c3f1ae8e00fa807130c17b13a8eb/app/src/main/java/org/fcitx/fcitx5/android/input/popup/PopupComponent.kt#L135-L138

it's never going to be "same size as key", I don't see any point of the name "FitAbove".